status/controls.py

```diff
@@ -0,0 +1,110 @@
+
+        # get information from databases
+        ws_data, ws_name_data = self.worksets_data()
+        all_control_data = {}
+        for control_data_type in ['negative', 'positive']:
+            all_control_data[control_data_type] = self.collect_control_info(control_data_type, ws_data, ws_name_data)
+
+        # define headers for controls.html
+        headers = [
+            ['Project', 'project'],
+            ['Sample ID', 'sample_id'],
+            ['Sample Name', 'customer_name'],
+            ['Sample Status', 'status_manual'],
+            ['Workset', 'workset_name'],
+            ['Workset Projects', 'workset_projects'],
+            ['Library Prep Status', 'prep_status'],
+            ['Flowcell(s)', 'sequenced_fc'],
+        ]
+
+        #anything in here is used to create the .html page. In essence, anything listed here can be accessed in /controls.html
+        self.write( 
+            t.generate(
+                gs_globals=self.application.gs_globals, user=self.get_current_user(),
+                all_control_data = all_control_data, # control_data is a dictionary with the control data, it can be called in the html with the name before the equal sign
+                headers=headers,
+                ws_data = ws_data,
+                lims_uri=BASEURI,
+            )
+        )
+
+    def find_control_data(self, control_type):
+        """Find control data from the couchDB from project/controls view
+        """
+        from collections import defaultdict
+        result = defaultdict(dict)
+        controls_view = self.application.projects_db.view("project/controls")
+
+        all_cont_proj = controls_view[[control_type,'']:[control_type,"Z"]]
+        for cont_proj in all_cont_proj:
+            for cont_sample in cont_proj.value:
+                for workset in cont_proj.value[cont_sample]: # here we create one entry in result for each workset, this will be one line in the controls table
+                    if workset != "no_workset":
+                        result[workset]["sample_id"] = cont_sample
+                        result[workset]["customer_name"] = cont_proj.value[cont_sample][workset]["customer_name"]
+                        if "status_manual" in cont_proj.value[cont_sample][workset]: # status originates from LIMS project overview, is often not set for controls
+                            result[workset]["status_manual"] = cont_proj.value[cont_sample][workset]["status_manual"]
+                        else:
+                            result[workset]["status_manual"] = "* In Progress" # asterisk indicates that the status in LIMS is not set, the sample has a workset and so MUST be at least "In Progress"
+                        result[workset]["project"] = cont_proj.key[1]
+                        result[workset]["workset_name"] = cont_proj.value[cont_sample][workset]["workset_name"]
+                        if not "workset_id" in cont_proj.value[cont_sample][workset]:
```

Comment:
Flip cases so that it checks for the positive condition, like in the prep_status `if` below?